packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/sdk/test/arm-test/src/static-helpers/urlTemplate.ts

```diff
@@ -0,0 +1,199 @@
+      }
+      vals.push(encodeComponent(val, reserved, op));
+      isFirst = false;
+    }
+  } else if (typeof value === "object") {
+    for (const key of Object.keys(value)) {
+      const val = value[key];
+      if (!isDefined(val)) {
+        continue;
+      }
+      // prepare the following parts: separator, key, value
+      vals.push(`${getFirstOrSep(op, isFirst)}`);
+      if (key) {
+        vals.push(`${encodeURIComponent(key)}`);
+        named && val === "" ? vals.push(ifEmpty) : vals.push("=");
+      }
+      vals.push(encodeComponent(val, reserved, op));
+      isFirst = false;
+    }
+  }
+  return vals.join("");
+}
+
+function getNonExpandedValue(option: ValueOptions) {
+  const { op, varName, varValue: value, isFirst, reserved } = option;
+  const vals: string[] = [];
+  const first = getFirstOrSep(op, isFirst);
+  const [named, ifEmpty] = getNamedAndIfEmpty(op);
+  if (named && varName) {
+    vals.push(encodeComponent(varName, reserved, op));
+    if (value === "") {
+      if (!ifEmpty) {
+        vals.push(ifEmpty);
+      }
+      return !vals.join("") ? undefined : `${first}${vals.join("")}`;
+    }
+    vals.push("=");
+  }
+
+  const items = [];
+  if (Array.isArray(value)) {
+    for (const val of value.filter(isDefined)) {
+      items.push(encodeComponent(val, reserved, op));
+    }
+  } else if (typeof value === "object") {
+    for (const key of Object.keys(value)) {
+      if (!isDefined(value[key])) {
+        continue;
+      }
+      items.push(encodeRFC3986URIComponent(key));
+      items.push(encodeComponent(value[key], reserved, op));
+    }
+  }
+  vals.push(items.join(","));
+  return !vals.join(",") ? undefined : `${first}${vals.join("")}`;
+}
+
+function getVarValue(option: ValueOptions): string | undefined {
+  const { op, varName, modifier, isFirst, reserved, varValue: value } = option;
+
+  if (!isDefined(value)) {
+    return undefined;
+  } else if (["string", "number", "boolean"].includes(typeof value)) {
+    let val = value.toString();
+    const [named, ifEmpty] = getNamedAndIfEmpty(op);
+    const vals: string[] = [getFirstOrSep(op, isFirst)];
+    if (named && varName) {
+      vals.push(encodeRFC3986URIComponent(varName));
+      val === "" ? vals.push(ifEmpty) : vals.push("=");
+    }
+    if (modifier && modifier !== "*") {
+      val = val.substring(0, parseInt(modifier, 10));
+    }
+    vals.push(encodeComponent(val, reserved, op));
+    return vals.join("");
+  } else if (modifier === "*") {
+    return getExpandedValue(option);
+  } else {
+    return getNonExpandedValue(option);
+  }
+}
+
+// ---------------------------------------------------------------------------------------------------
+// This is an implementation of RFC 6570 URI Template: https://datatracker.ietf.org/doc/html/rfc6570.
+// ---------------------------------------------------------------------------------------------------
+export function expandUrlTemplate(
```

Comment:
is there any uts to test against this specific operation?

Reply:
https://github.com/Azure/autorest.typescript/blob/a165ea8e19ea7e97148655787456bbc9be7b4053/packages/typespec-ts/test/modularUnit/static/urlTemplate.spec.ts